src/librustc_typeck/diagnostics.rs

```diff
@@ -1490,21 +1490,49 @@ For information on the design of the orphan rules, see [RFC 1023].
 
 E0118: r##"
 Rust can't find a base type for an implementation you are providing, or the type
-cannot have an implementation. For example, only a named type or a trait can
-have an implementation:
+cannot have an implementation. Erroneous code examples:
 
 ```
-type NineString = [char, ..9] // This isn't a named type (struct, enum or trait)
-impl NineString {
+type NineString = &'static [char];
+
+impl NineString { // error: no base type found for inherent implementation
     // Some code here
 }
+
+// or even more simple:
+impl TypeWhichDoesntExist { // error: no base type found for inherent implementation
+    // some code here
+}
 ```
 
-In the other, simpler case, Rust just can't find the type you are providing an
-impelementation for:
+To fix this error, please implement a trait on the type or declare a new one. Example:
 
 ```
-impl SomeTypeThatDoesntExist {  }
+type NineString = &'static [char];
```

Comment:
Type aliases don't change the situation at all, they're just confusing here since it _seems_ like you're suggesting them as a solution even if they won't work.